geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/ClientProxyMembershipID.java

```diff
@@ -176,149 +177,114 @@ public static ClientProxyMembershipID getClientId(DistributedMember member) {
+  public ClientProxyMembershipID(final DistributedMember member) {
+    this(null, 1, member);
     updateID(member);
   }
 
+  @VisibleForTesting
+  ClientProxyMembershipID(final byte[] identity, final int uniqueId,
+      final DistributedMember memberId) {
+    this.identity = identity;
+    this.uniqueId = uniqueId;
+    this.memberId = memberId;
+  }
 
   private transient String _toString;
 
-  // private transient int transientPort; // variable for debugging member ID issues
-
   @Override
   public String toString() {
-    if (this.identity != null
+    if (identity != null
         && ((InternalDistributedMember) getDistributedMember()).getMembershipPort() == 0) {
-      return this.toStringNoCache();
+      return toStringNoCache();
     }
-    if (this._toString == null) {
-      this._toString = this.toStringNoCache();
+    if (_toString == null) {
+      _toString = toStringNoCache();
     }
-    return this._toString;
+    return _toString;
   }
 
   /**
    * returns a string representation of this identifier, ignoring the toString cache
    */
   public String toStringNoCache() {
-    StringBuffer sb = new StringBuffer("identity(").append(getDSMembership()).append(",connection=")
-        .append(uniqueId);
+    StringBuilder sb =
+        new StringBuilder("identity(").append(getDSMembership()).append(",connection=")
+            .append(uniqueId);
     if (identity != null) {
       DurableClientAttributes dca = getDurableAttributes();
       if (dca.getId().length() > 0) {
-        sb.append(",durableAttributes=").append(dca).append(')').toString();
+        sb.append(",durableAttributes=").append(dca).append(')');
       }
     }
     return sb.toString();
   }
 
-  /**
-   * For Externalizable
-   *
-   * @see Externalizable
-   */
   @Override
   public void writeExternal(ObjectOutput out) throws IOException {
-    // if (this.transientPort == 0) {
-    // InternalDistributedSystem.getLogger().warning(
-    // String.format("%s",
-    // "externalizing a client ID with zero port: " + this.toString(),
-    // new Exception("Stack trace")));
-    // }
-    Assert.assertTrue(this.identity.length <= BYTES_32KB);
-    out.writeShort(this.identity.length);
-    out.write(this.identity);
-    out.writeInt(this.uniqueId);
-
-  }
+    if (identity.length > Short.MAX_VALUE) {
+      throw new IOException("HandShake identity length is too big");
+    }
 
-  /** returns the externalized size of this object */
-  public int getSerializedSize() {
-    return 4 + identity.length + 4;
+    out.writeShort(identity.length);
+    out.write(identity);
+    out.writeInt(uniqueId);
   }
 
-  /**
-   * For Externalizable
-   *
-   * @see Externalizable
-   */
   @Override
   public void readExternal(ObjectInput in) throws IOException, ClassNotFoundException {
     int identityLength = in.readShort();
-    if (identityLength > BYTES_32KB) {
-      throw new IOException(
-          "HandShake identity length is too big");
```

Comment:
comment: dead code. nice catch!

Reply:
Even more so, it was wrong code. Notice the original max constant was off by 1.

Reply:
yep!